### PR TITLE
[FileReferencesInstaller] Dont create build headers for frameworks

### LINKED ISSUE #4420: XCode 7.1 - Include of non-modular header inside framework
I have a project that was building fine that suddenly stopped building in xcode 7.1

The project has a development pod and also includes the GPUImage pod.  The development pod used to be able to reference GPUImage via:

````
#import <GPUImage/GPUImage.h>
````

But that now gives a compile error of "Include of non-modular header inside framework module"

I've tried setting the flag: 

````
CLANG_ALLOW_NON_MODULAR_INCLUDES_IN_FRAMEWORK_MODULES = 'YES'
````

but that makes no difference.

I tried to import GPUImage via:

````
@import GPUImage;
````

but that won't compile.  Is this a know issue ?

### PR DESCRIPTION
**Pull request by segiddins**
Closes #4420 .

- [x] CHANGELOG
- [x] Update integration specs

\c @neonichu @mtitolo

### CODE REVIEW

**Comment by mtitolo** [comment]
batman thumbs up

**Comment by mrackwitz** [comment]
Can you please summarize what the original issue was, how and why this PR solves the problem?
Because I'm not really convinced whether that's the right solution, but it's hard to follow the linked issue.

**Comment by segiddins** [comment]
The issue is that the search paths were being added to the headers directory in addition to the file references in the project, causing duplicates and Xcode to complain that there were non module includes taking place.

**Comment by neonichu** [comment]
LGTM :+1:

**Comment by segiddins** [comment]
Will merge when green

**Comment by segiddins** [comment]
:shipit:
